### PR TITLE
Move privacy enhancements docs from the GoQuorum repo

### PR DESCRIPTION
**Pull request by nicolae-leonte-go**
Move privacy enhancements documentation from the corresponding quorum branch.

### CODE REVIEW

File: docs/Reference/APIs/PrivacyEnhancementsAPIs.md
Comment (NicolasMassart) [inline docs/Reference/APIs/PrivacyEnhancementsAPIs.md:14]: Format tabs and code blocks using [the `=== "tab title"` format.](https://github.com/PegaSysEng/doc.common/wiki/MkDocs-And-Custom-Markdown-Guide#tabbed-code-blocks)

File: docs/Reference/APIs/PrivacyEnhancementsAPIs.md
Comment (NicolasMassart) [inline docs/Reference/APIs/PrivacyEnhancementsAPIs.md:16]: try not to mix different code in the same block. Here you have `curl` command which is a shell command call and the return value that can be displayed as json.

File: docs/Reference/APIs/PrivacyEnhancementsAPIs.md
Comment (NicolasMassart) [inline docs/Reference/APIs/PrivacyEnhancementsAPIs.md:12]: Use the `!!! example` [admonition panel.](https://github.com/PegaSysEng/doc.common/wiki/MkDocs-And-Custom-Markdown-Guide#admonition) for examples to be immediately visually recognised by the user.

File: docs/Reference/APIs/PrivacyEnhancementsAPIs.md
Comment (NicolasMassart) [inline docs/Reference/APIs/PrivacyEnhancementsAPIs.md:30]: Here this whole code example could be :

````markdown
#### Examples

!!! examples "JSON RPC example"

    === "Request"

        ```bash
        curl -X POST http://127.0.0.1:22000 --data '{"jsonrpc":"2.0","method":"eth_getContractPrivacyMetadata","params":["0x1932c48b2bf8102ba33b4a6b545c32236e342f34"],"id":15}' --header "Content-Type: application/json"
        ```

    === "Response"

        ```json
        {"jsonrpc":"2.0","id":15,"result":{"creationTxHash": [246,124,116,139,190,217,33,16,203,102,81,13,65,58,249,145,68,180,67,79,163,37,119,27,99,35,247,240,12,53,25,45,47,134,16,118,246,128,97,237,45,50,79,97,78,221,47,1 89,11,165,238,36,8,187,66,64,42,135,108,75,41,85,152,183],"privacyFlag":3}}
        ```

!!! examples "Geth console example"

    === "Request"

        ```javascript
        > eth.getContractPrivacyMetadata("0x1932c48b2bf8102ba33b4a6b545c32236e342f34");
        ```

    === "Response"

        ```json
        {
          creationTxHash: [246,124,116,139,190,217,33,16,203,102,81,13,65,58,249,145,68,180,67,79,163,37,119,27,99,35,247,240,12,53,25,45,47,134,16,118,246,128,97,237,45,50,79,97,78,221,47,1 89,11,165,238,36,8,187,66,64,42,135,108,75,41,85,152,183],
          privacyFlag: 3
        }
        ```
````

File: docs/Concepts/Privacy/PrivacyEnhancements.md
Comment (NicolasMassart) [inline docs/Concepts/Privacy/PrivacyEnhancements.md:129]: Add all abbreviations definitions at the end of the doc using the https://python-markdown.github.io/extensions/abbreviations/ syntax.

Use the `!!!note` admonition for notes.

```suggestion
!!! note
    If the EVM simulation impacts more than one contract, all contracts should have the same `PrivacyFlag`, otherwise the transaction is rejected.
    All contracts ACOTHs are included in the transmission and the Tessera node will create individual proofs (secure hash) for each ACOTH.**

*[PP]: Counter Party Protection
*[PSV]: Private State Validation
*[ACOTH]: Affected Contract's Original Transaction's encrypted payload Hash
*[SP]: Standard Private
*[EVM]: Ethereum Virtual Machine
*[P2P]: Peer to peer
```

**Review by chris-j-h — COMMENTED** [review]
https://docs.goquorum.consensys.net/en/latest/Concepts/Privacy/Privacy/#state-verification

This section of the Privacy Overview page could do with a line saying that state verification is possible with PE

File: docs/Concepts/Privacy/PrivacyEnhancements.md
Comment (nicolae-leonte-go) [reply docs/Concepts/Privacy/PrivacyEnhancements.md:14]: Once the abbreviations are added (you suggested a change for that as well - many thanks) would you say that the rest of the text should use just the abbreviations instead of the complete definitions? There are a couple of sentences where both are used.

File: docs/Concepts/Privacy/PrivacyEnhancements.md
Comment (nicolae-leonte-go) [reply docs/Concepts/Privacy/PrivacyEnhancements.md]: I suspect it is a reference to the actual APIs being invoked: /send or /receive (the rest service url)

File: docs/Reference/APIs/PrivacyEnhancementsAPIs.md
Comment (nicolae-leonte-go) [reply docs/Reference/APIs/PrivacyEnhancementsAPIs.md:18]: Updated the examples paragraph with your recommended formatting.

File: docs/Reference/APIs/PrivacyEnhancementsAPIs.md
Comment (nicolae-leonte-go) [reply docs/Reference/APIs/PrivacyEnhancementsAPIs.md]: Many thanks - looks much better.

File: docs/Reference/APIs/PrivacyEnhancementsAPIs.md
Comment (nicolae-leonte-go) [reply docs/Reference/APIs/PrivacyEnhancementsAPIs.md]: Updated the examples paragraph with your recommended formatting.

File: docs/Concepts/Privacy/PrivacyEnhancements.md
Comment (nicolae-leonte-go) [reply docs/Concepts/Privacy/PrivacyEnhancements.md:14]: I've applied your suggested change.

**Comment by nicolae-leonte-go** [comment]
> https://docs.goquorum.consensys.net/en/latest/Concepts/Privacy/Privacy/#state-verification
> 
> This section of the Privacy Overview page could do with a line saying talking about PE

I've added a reference to privacy enhancements to the State verification paragraph.

**Review by NicolasMassart — APPROVED** [review]
LGTM once the few markdown and guideline fixes (details) are done.
See each failing test details for more infos on the things to fix.

**Comment by NicolasMassart** [comment]
@nicolae-leonte-go I will let you pass it in "ready for review" status otherwise I would take the risk to merge something while you still have things you want to change on it. Once ready for review I will merge. It's the 🟢  signal.

**Comment by NicolasMassart** [comment]
@nicolae-leonte-go any issues on this one? Just let me know if you need something from me.

**Pull request merged**
